R/group_data.R

```diff
@@ -63,3 +63,21 @@ add_sum_of_views <- function(df, view_name_pattern, method, new_name = view_name
   df_plus_new_group <- bind_rows(df, new_group)
   return(df_plus_new_group)
 }
+
+#' @param df data.frame of daily unique pageviews and page paths with site ids for noms
+#'           expects pagepath, date, uniquePageviews and pageviews
+#' @return data frame with date, uniquePageviews and site_no
+group_by_site_id <- function(df) {
+  
+  date <- df$date[1]
+  
+  df <- df %>%
+    mutate(site_no = regmatches(df$pagePath, gregexpr("[[:digit:]]+", df$pagePath))) %>% #get all numeric values into new column
+    unnest(site_no) %>% #unlist list column
```

Comment:
this works similarly to separate_rows, and breaks up the site_no column if it has several items in a list like c("site_no","site_no")